emergence_lib/src/signals/equation.rs

```diff
@@ -0,0 +1,163 @@
+use std::{collections::VecDeque, f32::consts};
+
+use bevy::prelude::trace;
+
+use crate::simulation::geometry::TilePos;
+
+use super::{SignalStrength, DECAY_RATE, DECAY_THRESHOLD, DIFFUSIVITY};
+
+#[derive(Clone, Copy, Debug)]
+pub struct SignalEmission {
+    /// Time at which the signal was emitted.
+    pub time: f32,
+    /// Strength of the emitted signal.
+    pub strength: f32,
+    /// Position of the signal source.
+    pub source: TilePos,
+}
+
+/// A solver for the diffusion equation.
+///
+/// The diffusion equation can be formulated in its generic form as:
+///
+/// ∂ϕ(x, t)/∂t = ∇⋅(D(ϕ, x) ∇ϕ(x, t))
+///
+/// where:
+/// - ϕ is the density of the diffusing material. In emergence's case, ϕ is the signal strength.
+/// - D is the diffusivity of the environment. For emergence, this is a constant for now, making
+///   our case equivalent to the classic [heat equation](https://en.wikipedia.org/wiki/Heat_equation).
+/// - x is a location.
+/// - t is the time.
+/// - ∇⋅ is the divergence operator.
+/// - ∇ is the gradient operator.
+///
+/// To solve the equation, one can use the [fundamental solution](https://en.wikipedia.org/wiki/Heat_equation#Fundamental_solutions)
+/// of the heat equation, in 2D:
+///
+/// ϕ(x, t) = 1/(4 π D t) * exp(−x⋅x / (4 D t))
+///
+/// where x⋅x is a dot product.
+///
+/// The fundamental solution solves a problem with no boundary conditions and
+/// an initial density given by the [Dirac delta function](https://en.wikipedia.org/wiki/Dirac_delta_function).
+/// The general solution, for any initial condition is a convolution with the fundamental solution:
+///
+/// s(x, t) = ∫ Φ(x−y, t) g(y) dy
+///
+/// where:
+/// - g is the initial condition, i.e. s(x, 0).
+/// - y is a surface element in the same space as x.
+///
+/// Practically, we don't want to deal with costly convolutions, so we'll try to avoid them.
+/// Because signals are emitted in pulses, and convolution is distributive, we can separate each
```

Comment:
Do we *want* signals to be implemented in pulses? I think ideally these would be better modeled as plateaus, to avoid situations where wave fronts propagate away from the actual source.